bauble/src/types/path.rs

```diff
@@ -76,18 +76,22 @@ impl<S: AsRef<str>> TypePathElem<S> {
         }
     }
 
+    /// Convert this to an owned path element.
     pub fn to_owned(&self) -> TypePathElem<String> {
         TypePathElem(self.0.to_owned())
     }
 
+    /// Borrow this path element.
     pub fn borrow(&self) -> TypePathElem<&str> {
         TypePathElem(self.0.borrow())
     }
 
+    /// For path elements this is always false.
     pub fn is_empty(&self) -> bool {
         false
     }
 
+    /// For path elements this is always one.
     pub fn len(&self) -> usize {
         1
     }
```

Comment:
Why do we need these? Is there a possibility of these not always being false/1 in the future? The only other reason I can think of to have these functions is if we have a macro somewhere that's doing something like `$path.len()` and just hoping it works, but I can't imagine why we'd do that in Rust when we have traits...

Reply:
I think its more that these help define what a `TypePathElem` is. And even if I remove these you could do `path_elem.len()` as it derefs to a `TypePath`. And these methods overshadow those and are much more efficient.